File: app/src/main/java/io/agora/chatdemo/sign/SplashViewModel.java
package io.agora.chatdemo.sign;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import io.agora.chatdemo.general.net.Resource;
import io.agora.chatdemo.general.repositories.EMClientRepository;

public class SplashViewModel extends AndroidViewModel {
    private EMClientRepository mRepository;

    public SplashViewModel(@NonNull Application application) {
        super(application);
        mRepository = new EMClientRepository();
    }

    public LiveData<Resource<Boolean>> getLoginData() {
        return mRepository.loadAllInfoFromHX();
    }
}

File: app-kotlin/src/main/kotlin/io/agora/chatdemo/page/splash/viewModel/SplashViewModel.kt
package io.agora.chatdemo.page.splash.viewModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import io.agora.chatdemo.page.splash.repository.ChatClientRepository
import kotlinx.coroutines.flow.flow

class SplashViewModel(application: Application) : AndroidViewModel(application) {
    private val mRepository: ChatClientRepository = ChatClientRepository()

    fun loginData() = flow {
        emit(mRepository.loadAllInfoFromAgora())
    }

}